bin/node/src/commands/node.rs

```diff
@@ -84,13 +85,29 @@ impl NodeCommand {
         let keypair = libp2p_identity::Keypair::secp256k1_from_der(&mut private_key.0)
             .map_err(|_| anyhow::anyhow!("Failed to parse private key"))?;
 
+        let l1_eth_rpc = self
+            .l1_eth_rpc
+            .clone()
+            .ok_or(anyhow::anyhow!("Missing L1 RPC URL. Please specify with `--l1`."))?;
+        let l1_beacon = self
+            .l1_beacon
+            .clone()
+            .ok_or(anyhow::anyhow!("Missing L1 beacon URL. Please specify with `--l1.beacon`."))?;
+        let l2_provider_rpc = self.l2_provider_rpc.clone().ok_or(anyhow::anyhow!(
+            "Missing L2 provider RPC URL. Please specify with `--l2.provider`."
+        ))?;
+        let l2_engine_rpc = self
+            .l2_engine_rpc
+            .clone()
+            .ok_or(anyhow::anyhow!("Missing L2 engine RPC URL. Please specify with `--l2`."))?;
```

Comment:
We should be letting `clap` handle this IMO - non-optional flags should already have a nice "flag `x` is required" type message.